app/sidekiq/worker_job.rb

```diff
@@ -0,0 +1,19 @@
+class WorkerJob
+  include Sidekiq::Job
+
+  def perform
+    yesterday_rank = Rank.create(date: Date.yesterday)
+
+    books = Book.all
+    books.each do |book|
+      view = Rails.cache.fetch("views/#{book.id}") { 0 }
+      BookRank.create(book: book, rank: yesterday_rank, view: view)
```

Comment:
We can also wrap all there queries under a transaction, so they will be rollbacked whenever error happens in the middle of the processs.

ref: https://api.rubyonrails.org/classes/ActiveRecord/Transactions/ClassMethods.html